R/data.table.R

```diff
@@ -1806,7 +1816,8 @@ replace_dot_alias = function(e) {
           }
         } else GForce = .gforce_ok(jsub)
         gforce_jsub = function(x, names_x) {
-          x[[1L]] = as.name(paste0("g", x[[1L]]))
+          call_name <- if (is.symbol(x[[1L]])) x[[1L]] else x[[1L]][[3L]] # latter is like data.table::shift, #5942. .gshift_ok checked this will work.
```

Comment:
```suggestion
          call_name = if (is.symbol(x[[1L]])) x[[1L]] else x[[1L]][[3L]] # latter is like data.table::shift, #5942. .gshift_ok checked this will work.
```